streamlit_web.py

```diff
@@ -283,6 +300,24 @@ def overall_freq(freq_range):
         )
     )
 
+    responses_end = len(main_df.columns) - 3
+    responses_df = main_df[main_df.columns[1:responses_end]]
+    responses_df.replace("", "NA")
+
+    # concatenate all words into normalized string and make into wordcloud
+    words = az.concatenate(responses_df)
+    cloud_stopwords = set(STOPWORDS)
+    wordcloud = (WordCloud(width = 800, height = 800,
+                    background_color = 'white',
+                    stopwords = cloud_stopwords,
+                    min_font_size = 10).generate(words))
+
+    # plot wordcloud by temporarily savings as a file and displaying
+    wordcloud.to_file("resources/images/word_cloud.png")
+    st.image("resources/images/word_cloud.png")
+    os.remove("resources/images/word_cloud.png")
```

Comment:
Paths are hardcoded.

Reply:
I can't find any alternative method that displays the wordcloud in streamlit, and saving and displaying an image of the wordcloud seems to be how other people online display it on streamlit. Would there an alternative way to save and display an image that isn't hardcoded?